### PR TITLE
✨Source XKCD: Migrate Python CDK to Low code

### PR DESCRIPTION
**Pull request by btkcodedev**
## What
*Migrating Source XKCD to Low-Code CDK*

## How
*Developed using (Configuration Based Source) low-code CDK*

## Recommended reading order
1. `spec.yaml`
2. `manifest.yaml`
3. `schemas/*`

## 🚨 User Impact 🚨
- Migration to low-code

### CODE REVIEW

File: airbyte-integrations/connectors/source-xkcd/integration_tests/acceptance.py [hunk 1]
```diff
@@ -10,4 +11,6 @@
 @pytest.fixture(scope="session", autouse=True)
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-xkcd/integration_tests/acceptance.py:11]: @btkcodedev just delete the whole file.

In fact, if there are NO python custom components, can you delete all python files, and move `manifest.yaml` to top level? Just leave the following:
- `metadata.yaml`
- `manifest.yaml`
- docs
- README (we'll need to rewrite it)

File: airbyte-integrations/connectors/source-xkcd/metadata.yaml [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
   definitionId: 80fddd16-17bd-4c0c-bf4a-80df7863fc9d
-  dockerImageTag: 0.1.10
+  dockerImageTag: 0.2.0
   dockerRepository: airbyte/source-xkcd
   githubIssueLabel: source-xkcd
   icon: xkcd.svg
   license: MIT
-  name: xkcd
+  name: Xkcd
+  releaseDate: TODO
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-xkcd/metadata.yaml:14]: Set the original release date of the connector if you're going for it here.

File: airbyte-integrations/connectors/source-xkcd/metadata.yaml [hunk 3]
```diff
@@ -21,7 +25,7 @@ data:
   documentationUrl: https://docs.airbyte.com/integrations/sources/xkcd
   tags:
     - language:python
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-xkcd/metadata.yaml:27]:
```suggestion
    - language:python
    - language:manifest-only
```

Only apply if you choose to delete everything except manifest+metadata.

File: airbyte-integrations/connectors/source-xkcd/pyproject.toml [hunk 4]
```diff
@@ -3,26 +3,25 @@ requires = [ "poetry-core>=1.0.0",]
...
 homepage = "https://airbyte.com"
 repository = "https://github.com/airbytehq/airbyte"
-[[tool.poetry.packages]]
-include = "source_xkcd"
+packages = [ { include = "source_xkcd" }, {include = "main.py" } ]
 
 [tool.poetry.dependencies]
 python = "^3.9,<3.12"
-airbyte-cdk = "0.83.0"
+airbyte-cdk = "^2"
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-xkcd/pyproject.toml:19]: Current CDK version is 3.9 btw. But, delete the whole file, let's manifest-only this.

File: airbyte-integrations/connectors/source-xkcd/metadata.yaml [hunk 5]
```diff
@@ -1,13 +1,17 @@
 data:
+  allowedHosts:
+    hosts:
+      - xkcd.com
   connectorSubtype: api
   connectorType: source
   definitionId: 80fddd16-17bd-4c0c-bf4a-80df7863fc9d
-  dockerImageTag: 0.1.10
+  dockerImageTag: 0.2.0
   dockerRepository: airbyte/source-xkcd
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-xkcd/metadata.yaml:9]: This is also missing the block with build options → base image. Look up what is the latest `airbyte/source-declarative-manifest:3.9.0` sha hash is, and put the whole block here. I can look tomorrow, but it's 10pm and my brain is fried today. Let's make this a manifest-only connector!

**Review by natikgadzhi — COMMENTED** [review]
Let's migrate this all the way to manifest-only, left a bunch of comments.

File: airbyte-integrations/connectors/source-xkcd/metadata.yaml (hunk 5 above)
Comment (btkcodedev) [reply airbyte-integrations/connectors/source-xkcd/metadata.yaml:9]: Yes sure!

**Comment by btkcodedev** [comment]
CI is green in the normal way now, lets see about the manifest only way
image

**Comment by btkcodedev** [comment]
Ok it seems like connector tests are not configured for manifest-only connectors
Checks from my side,
1. Tested from the builder:
 - Stream Fetch :white_check_mark: 
 - Pagination :white_check_mark: 
 - Builder Compatible (The pagination cursor has some workaround as it's injected in the path with increment so it only works in Yaml format) :white_check_mark: 
 
image

**Comment by btkcodedev** [comment]
Ok it seems like connector tests are not configured for manifest-only connectors
Checks from my side,
1. Tested from the builder:
 - Stream Fetch :white_check_mark: 
 - Pagination :white_check_mark: 
 - Builder Compatible (The pagination cursor has some workaround as the incremental value is injected in the path, so it only works in Yaml format) :white_check_mark: 
 
image

**Comment by btkcodedev** [comment]
Ok it seems like connector tests are not configured for manifest-only connectors
Checks from my side,
1. Tested from the builder:
 - Stream Fetch :white_check_mark: 
 - Pagination :white_check_mark: 
 - Builder Compatible :white_check_mark: 
  (Note: The pagination cursor has some workaround where builder UI is not supported) 
  Works as whole
  
  config:
  { comic_number: 2600 } # Optional
image

**Comment by btkcodedev** [comment]
Done & Done - Review left for merge

**Comment by natikgadzhi** [comment]
@alafanechere @bnchrch that's a good concern — we need to figure out and run tests on manifest-only connectors.

File: airbyte-integrations/connectors/source-xkcd/manifest.yaml [hunk 6]
```diff
@@ -0,0 +1,134 @@
+version: 2.0.0
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-xkcd/manifest.yaml:1]:
```suggestion
version: 3.10.2
```

Comment (natikgadzhi) [reply airbyte-integrations/connectors/source-xkcd/manifest.yaml:1]: YOLO

**Review by natikgadzhi — APPROVED** [review]
Mostly great! I've rewritten the README to reflect that this is our first manifest-only connector. Since it's so lightweight, I'm okay merging.

**Pull request merged**